javarosa/src/main/java/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -687,22 +673,16 @@ public static Double toNumeric(Object o) {
         } else if (o instanceof Double) {
             val = (Double)o;
         } else if (o instanceof String) {
-            /* annoying, but the xpath spec doesn't recognize scientific notation, or +/-Infinity
-             * when converting a string to a number
-             */
-
-            String s = (String)o;
+            String s = ((String)o).trim();
             double d;
             try {
-                s = s.trim();
-                for (int i = 0; i < s.length(); i++) {
-                    char c = s.charAt(i);
-                    if (c != '-' && c != '.' && (c < '0' || c > '9'))
-                        throw new NumberFormatException();
+                checkForInvalidNumericCharacters(s);

```

Comment:
Somewhat awkwardly it's probably for the best if this is "checkforInvalidNumericOrDateStringCharacters()" since both are necessary for the code below to make sense. Otherwise someone is likely to think the date checking is out of order.